### PR TITLE
feat: ✨ add mutex map

### PR DESCRIPTION
**Pull request by thanhpp**

### CODE REVIEW

File: pkg/ds/mutexmap/mutexmap.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+func NewMutexMap[K comparable, V any]() MutexMap[K, V] {
+	return MutexMap[K, V]{
+		data: make(map[K]V),
+	}
+}
+
+func (m *MutexMap[K, V]) Store(k K, v V) {
+	m.l.Lock()
+	m.data[k] = v
+	m.l.Unlock()
```
Comment (onlylove97) [inline pkg/ds/mutexmap/mutexmap.go:19]: chuyển defer unlock lên trên đi e :v

File: pkg/ds/mutexmap/mutexmap.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+func (m *MutexMap[K, V]) Load(k K) (v V, ok bool) {
+	m.l.RLock()
+	defer m.l.RUnlock()
+	v, ok = m.data[k]
```
Comment (secmask) [inline pkg/ds/mutexmap/mutexmap.go:25]: can be `return v, ok`

Comment (thanhpp) [reply pkg/ds/mutexmap/mutexmap.go:31]: updated in https://github.com/KyberNetwork/tradinglib/pull/4/commits/d5bc0e5289d25321cdc0579b8d914558b8d5a7ca

**Review by ngocthanh1389 — APPROVED** [review]

**Comment by ngocthanh1389** [comment]
@thanhpp should name "lockedmap" is better, mutex map mean a map of mutex

**Pull request merged**